src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -107,6 +108,15 @@ private async Task<int> PromoteBuildAsync(Build build, Channel targetChannel, IR
                 return Constants.ErrorCode;
             }
 
+            var (arcadeSDKSourceBranch, arcadeSDKSourceSHA) = await GetSourceBranchInfoAsync(build).ConfigureAwait(false);
+
+            // This condition can happen when for some reason we failed to determine the source branch/sha 
+            // of the build that produced the used Arcade SDK or when the user specify an invalid combination
+            // of source-sha/branch parameters.
+            if (arcadeSDKSourceBranch == null && arcadeSDKSourceSHA == null)
+            {
```

Comment:
What's the error message if this ahppens?

Reply:
The cases where this happens the message is printed by `GetSourceBranchInfoAsync`.